didcomm/secrets/secrets_resolver.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from didcomm.common.types import JWK
+
+
+class SecretsResolver(ABC):
+    """Resolves secrets such as private keys to be used for signing and encryption."""
+
+    @abstractmethod
+    async def get_key(self, kid: str) -> JWK:
+        """
+        A private key identified by the given key ID.
+
+        :raises NoJWKKeyException: if there is no key for the given key ID
+        :param kid: the key ID identifying a private key
+        :return: a private key in JWK format
+        """
+        pass
+
+    @abstractmethod
+    async def get_keys(self, did: str) -> List[JWK]:
+        """
+        All private keys for the given DID
+
+        :param did: the DID get all private keys for
+        :return: a possible empty list of all private keys for the given DID in JWK format
+        """
+        pass
+
+
+def register_default_secrets_resolver(secrets_resolver: SecretsResolver):
```

Comment:
I would prefer to avoid global state

Reply:
It's possible to pass dedicated instances of resolvers into all methods dealing with pack/unpack. So, if a caller prefers to avoid a global state, it can be done via standard API.
However, I believe it can be quite common and fine in many cases to register some default resolvers, so that there is no need to pass the same instances into all pack/unpack all the time.